data/pigui/modules/radar.lua

```diff
@@ -305,26 +390,67 @@ local function displayRadar()
 	end
 
 	-- Handle mouse if it is in the radar area
-	local mp = ui.getMousePos()
-	-- TODO: adjust properly for 3D radar; bit more challenging as its an ellipse
-	if (mp - center):length() < radar2d.getRadius(radar2d) then
+	local isMouseOverRadar = function()
+		if shouldDisplay2DRadar then
+			local mp = ui.getMousePos()
+			return (mp - center):length() < radar2d.getRadius(radar2d)
+		end
+		return ui.isMouseHoveringRect(
+			Vector2(center.x - ui.reticuleCircleRadius * 0.9,
+			        center.y - ui.reticuleCircleRadius * 0.7),
+			Vector2(center.x + ui.reticuleCircleRadius * 0.9,
+			        center.y + ui.reticuleCircleRadius * 0.7))
+	end
+	if radar_popup_displayed or isMouseOverRadar() then
 		ui.popup("radarselector", function()
 			if ui.selectable(lui.HUD_2D_RADAR, shouldDisplay2DRadar, {}) then
-				toggle_radar = true
+				if not shouldDisplay2DRadar then
+					toggle_radar = true
+				end
+				radar_popup_displayed = false
 			end
 			if ui.selectable(lui.HUD_3D_RADAR, not shouldDisplay2DRadar, {}) then
-				toggle_radar = true
+				if shouldDisplay2DRadar then
+					toggle_radar = true
+				end
+				radar_popup_displayed = false
+			end
+		end)
```

Comment:
Q: Now that we have the icon button to switch the radar between 2d and 3d modes, do we still want to support the right-click popup?  The code is a bit messy and I don't really see the utility.

Reply:
Muscle-memory for right-click popups is pretty poor anyway..

Thanks, so unless someone else is going to step in with a strong argument "for" I'll remove it.

Reply:
Removed in e717c283c9012a11f037e919848e2ea590ea6242